src/fluent_compiler/ast_compat.py

```diff
@@ -58,12 +57,29 @@ def NewAst(...):
 Call = ast.Call
 FunctionDef = ast.FunctionDef
 Name = ast.Name
-NameConstant = ast.NameConstant
 Try = ast.Try
 arg = ast.arg
 keyword = ast.keyword
 
 
+if sys.version_info >= (3, 8):
+    Constant = ast.Constant
+else:
+    # For Python 3.6/3.7, in terms of runtime behaviour we could also use
+    # Constant for Str/Num, but this seems to trigger bugs when decompiling with
+    # ast_decompiler, which is needed by tests. So we use the more normal
+    # ast that Python 3.6/3.7 use for this code.
+    def Constant(arg, **kwargs):
+        if isinstance(arg, str):
+            return ast.Str(arg, **kwargs)
+        elif isinstance(arg, (int, float)):
+            return ast.Num(arg, **kwargs)
+        elif arg is None:
+            return ast.NameConstant(arg, **kwargs)
+        else:
+            raise NotImplementedError(f"Constant not implemented for args of type {type(arg)}")
+
+
 def traverse(ast_node, func):
```

Comment:
May be move `traverse` to the `codegen` module? `traverse` is not a part of the `ast` API.

Reply:
> May be move `traverse` to the `codegen` module? `traverse` is not a part of the `ast` API.

Yes, that would be good. Feel free to add a commit to this PR.

Reply:
Actually I realised you won't be able to push to my branch, please feel free to do a follow up PR, or address it as part of other fixes.